b2luigi/batch/processes/htcondor.py

```diff
@@ -0,0 +1,225 @@
+        if not match:
+            raise RuntimeError("Batch submission failed with output " + output)
+
+        self._batch_job_id = int(match.group(0)[:-1])
+
+    def kill_job(self):
+        if not self._batch_job_id:
+            return
+        subprocess.check_call(["condor_rm", str(self._batch_job_id)], stdout=subprocess.DEVNULL)
+
+    def _create_htcondor_submit_file(self):
+        output_path = get_output_dirs(self.task, create_folder=True)
+        submit_file_path = os.path.join(output_path, "job.submit")
+        executable_wrapper_path = self._create_executable_wrapper()
+
+        log_file_dir = get_log_file_dir(self.task)
+        stderr_log_file = log_file_dir + "job.err"
+        stdout_log_file = log_file_dir + "job.out"
+        job_log_file = log_file_dir + "job.log"
+
+        general_settings = get_setting("htcondor_settings", dict())
+
+        try:
+            job_settings = self.task.htcondor_settings
+        except AttributeError:
+            job_settings = {}
+
+        with open(submit_file_path, "w") as submit_file:
+            submit_file.write(f"executable = {executable_wrapper_path}\n")
+            submit_file.write("should_transfer_files = YES\n")
+            submit_file.write("transfer_input_files = settings.json\n")
+            submit_file.write(f"log = {job_log_file}\n")
+            submit_file.write(f"output = {stdout_log_file}\n")
+            submit_file.write(f"error = {stderr_log_file}\n")
+
+            for key, item in general_settings.items():
+                submit_file.write(f"{key} = {item}\n")
+
+            for key, item in job_settings.items():
+                submit_file.write(f"{key} = {item}\n")
+
+            submit_file.write(f"queue 1\n")
+
+        return submit_file_path
+
+    def _create_executable_wrapper(self):
+
+        env_setup_path = get_setting("env_setup")
+
+        if not os.path.isfile(env_setup_path):
+            raise FileNotFoundError(f"Environment setup script {env_setup_path} does not exist.")
+
+        output_path = get_output_dirs(self.task, create_folder=True)
+        shell = get_setting("shell", "bash")
+
+        executable_wrapper_path = os.path.join(output_path, "executable_wrapper.sh")
+
+        condor_executable_cmd, _ = self._create_condor_executable_cmd()
+        condor_executable_cmd = " ".join(condor_executable_cmd)
+        with open(executable_wrapper_path, "w") as exec_wrapper:
+            exec_wrapper.write(f"#!/bin/{shell}\n")
+            exec_wrapper.write(f"source {env_setup_path}\n")
+            exec_wrapper.write("echo 'Starting executable'\n")
+            exec_wrapper.write("cd /jwd\n")
+            exec_wrapper.write(f"{condor_executable_cmd}\n")
+
+        # make wrapper executable
+        st = os.stat(executable_wrapper_path)
+        os.chmod(executable_wrapper_path, st.st_mode | stat.S_IEXEC)
+
+        return executable_wrapper_path
+
+    def _create_condor_executable_cmd(self):
+
```

Comment:
I think there is already a function like this, or? Maybe we can merge them?

Reply:
In principle it's the same function. In the original function, the current python executable is used to execute the task on the worker node (in the LSF case). At the NAF I got some errors with the python libraries (I could try to reproduce them, but I assume that  the OS can differ from the submission and the worker node, at the NAF both SL6 and CentOS7 are used on the worker nodes if the OS is not specified). So this function is a workaround by using the current python3 command after setting up the environment. I suppose that dealing with the job environment is the biggest problem at the moment. I could read up a little and rethink the approach of providing the correct one.

Reply:
I guess your approach is fine, so we can keep it. I was more concerned about having two functions with more or less the  same code. So maybe it is possible to have an argument to the original function, so it can be reused as much as possible.

Reply:
I added an default argument to the `create_cmd_from_task` function to avoid code duplication. This function has been removed in commit f96812e

Reply:
Nice!